### PR TITLE
HDDS-10013. Clean up test dependencies

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

1. d02d6b6476: Provide the same set of basic test dependencies for all modules via `hadoop-hdds/hadoop-dependency-test`.
2. cfae55296a: Group test dependencies in submodule POMs (after prod. dependencies), and order them alphabetically.
3. Move some dependency definitions to root POM's `dependencyManagement`:
   * 9f824f3d67: move `org.bouncycastle` from `hadoop-hdds/pom.xml`
   * 010981f754: move jooq and spring dependency definitions from Recon
4. Remove unused dependencies:
   * ead0b9c4e7: remove unused JUnit4 dependency
   * 9c2dd4cdec: remove JUnit4 dependency inherited by all hdds modules
   * 4019c405a8: remove unused mockito1/powermock dependencies
   * b467bfc03f: remove leftover migration support
   * 6ed09c055d: remove leftover junit-vintage-engine
5. Misc. improvements:
   * 99cd66efcb: add explicit test scope in sub-module POMs
   * 55265083bd: remove test scope from `dependencyManagement` in root POM
   * 346689b4fc: depend on `hamcrest-all` consistently (some modules used `hamcrest-core`, some `hamcrest-all`, some both)
   * 76e4a612d6: fix indent (mostly in `hadoop-hdds/rocks-native/pom.xml`)

https://issues.apache.org/jira/browse/HDDS-10013

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/7342330195

### CODE REVIEW

File: hadoop-hdds/rocks-native/pom.xml
Comment (smengcl) [inline hadoop-hdds/rocks-native/pom.xml:1]: Tagging @swamirishi just in case he'd catch something

**Review by smengcl — APPROVED** [review]
lgtm

File: hadoop-hdds/rocks-native/pom.xml
Comment (adoroszlai) [reply hadoop-hdds/rocks-native/pom.xml:1]: Thanks @smengcl for the review.  With `--ignore-space-change` the diff is much smaller there:

```diff
diff --git hadoop-hdds/rocks-native/pom.xml hadoop-hdds/rocks-native/pom.xml
index c9480ccbb2..d3a015a18f 100644
--- hadoop-hdds/rocks-native/pom.xml
+++ hadoop-hdds/rocks-native/pom.xml
@@ -25,10 +25,6 @@
   <artifactId>hdds-rocks-native</artifactId>
 
   <dependencies>
-        <dependency>
-            <groupId>org.apache.ozone</groupId>
-            <artifactId>hdds-test-utils</artifactId>
-        </dependency>
     <dependency>
       <groupId>org.apache.ozone</groupId>
       <artifactId>hdds-managed-rocksdb</artifactId>
@@ -37,19 +33,16 @@
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-io</artifactId>
     </dependency>
+
+    <!-- Test dependencies -->
     <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-api</artifactId>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-params</artifactId>
+      <groupId>org.apache.ozone</groupId>
+      <artifactId>hdds-hadoop-dependency-test</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-junit-jupiter</artifactId>
+      <groupId>org.apache.ozone</groupId>
+      <artifactId>hdds-test-utils</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>
```

**Comment by adoroszlai** [comment]
Thanks @smengcl for the review.

**Pull request merged**